### PR TITLE
Fix frontend CLI flags

### LINKED ISSUE #1774: Mismatch between the yaml config and the CLI flags in Frontend config
`max-outstanding-requests-per-tenant` and `compress-http-responses` CLI flags are associated with querier, it should be with frontend as the YAML file takes config from `frontend` block.
https://github.com/cortexproject/cortex/blob/master/pkg/querier/frontend/frontend.go#L52-L53

### PR DESCRIPTION
**Pull request by achilles42**
Signed-off-by: Praveen Shukla <praveen.shukla.c42@gmail.com>

**What this PR does**:
Move `max-outstanding-requests-per-tenant` and `compress-http-responses` CLI flags from `querier` to `frontend`.

**Which issue(s) this PR fixes**:
Fixes #1774 

**Checklist**
- [ ] Tests updated
- [X] Documentation added
- [X] `CHANGELOG.md` updated

cc @pracucci

### CODE REVIEW

File: docs/arguments.md [hunk 1]
```diff
@@ -61,8 +61,6 @@ The ingester query API was improved over time, but defaults to the old behaviour
 
    Time since the last sample after which a time series is considered stale and ignored by expression evaluations.
 
-## Query Frontend
-
 - `-querier.align-querier-with-step`
```
Comment (gouthamve) [inline docs/arguments.md]: All the flags below are also part of the frontend and not the querier. They should be updated in `pkg/querier/queryrange/roundtrip.go`

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,7 @@
 * [CHANGE] The frontend component has been refactored to be easier to re-use. When upgrading the frontend, cache entries will be discarded and re-created with the new protobuf schema. #1734
 * [CHANGE] Remove direct DB/API access from the ruler
 * [CHANGE] Removed `Delta` encoding. Any old chunks with `Delta` encoding cannot be read anymore. If `ingester.chunk-encoding` is set to `Delta` the ingester will fail to start. #1706
+* [CHANGE] Move frontend CLI flag from querier to frontend #1774
```
Comment (gouthamve) [inline CHANGELOG.md]: Should also mention which flags were updated to what.

**Comment by pracucci** [comment]
@achilles42 Do you still have interest into this PR? If so, I would suggest you to reiterate on @gouthamve feedback, otherwise we could close the PR.

**Comment by pracucci** [comment]
_I'm going to close this PR due to inactivity. Feel free to reopen it in case you're still interested into working on this._

**Pull request closed without merging**